src/Repository/OrderingRepository.php

```diff
@@ -0,0 +1,260 @@
+            ->from(BkOrdering::class, 'bo');
+        if ($isoCode2 !== null) {
+            $qb->join(BkCountries::class, 'c', 'WITH', 'bo.country_id = c.country_id')
+                ->where('c.iso_code_2 = :isoCode2')
+                ->setParameter('isoCode2', $isoCode2);
+        } else {
+            $qb->where('bo.country_id IS NULL');
+        }
+
+        return $qb->getQuery()->getOneOrNullResult();
+    }
+
+    public function updateOrdering($value, $countryId = null)
+    {
+        if (is_string($value)) {
+            $value = json_decode($value, true);
+            if (json_last_error() !== JSON_ERROR_NONE) {
+                // Handle JSON decode error
+                return false;
+            }
+        }
+
+        $idArray = array_column($value, 'id');
+        $value = json_encode($idArray);
+
+        if ($countryId !== null) {
+            return $this->updateOrderingForSpecificCountry($value, $countryId);
+        }
+
+        return $this->updateOrderingForNullCountry($value);
+    }
+
+    private function updateOrderingForSpecificCountry($value, $countryId)
+    {
+        $countryRepo = $this->_em->getRepository(BkCountries::class);
+        $country = $countryRepo->findOneBy(['country_id' => $countryId]);
+
+        if (!$country) {
+            return false;
+        }
+
+        $orderingRepo = $this->_em->getRepository(BkOrdering::class);
+        $ordering = $orderingRepo->findOneBy(['country_id' => $country->getCountryId()]);
+
+        if (!$ordering) {
+            return false;
+        }
+
+        $ordering->setValue($value);
+        $this->_em->persist($ordering);
+        $this->_em->flush();
+
+        return true;
+    }
+
+    private function updateOrderingForNullCountry($value)
+    {
+        $orderingRepo = $this->_em->getRepository(BkOrdering::class);
+        $ordering = $orderingRepo->findOneBy(['country_id' => null]);
+
+        if (!$ordering) {
+            return false;
+        }
+
+        $ordering->setValue($value);
+        $this->_em->persist($ordering);
+        $this->_em->flush();
+
+        return true;
+    }
+
+    public function getOrdering(?string $isoCode2)
+    {
+        $ordering = $this->findOneByCountryIsoCode($isoCode2);
+
+        if (empty($ordering)) {
+            return null;
+        }
+
+        $result = [
+            'id' => $ordering->getId(),
+            'country_id' => $ordering->getCountryId(),
+            'value' => [],
+            'createdAt' => $ordering->getCreatedAt(),
+            'updatedAt' => $ordering->getUpdatedAt(),
+            'status' => true,
+        ];
+
+        $paymentMethodIds = is_string($ordering->getValue())
+            ? json_decode($ordering->getValue(), true)
+            : $ordering->getValue();
+
+        foreach ($paymentMethodIds as $id) {
+            $paymentMethodRepo = $this->_em->getRepository(BkPaymentMethods::class);
+            $paymentMethodData = $paymentMethodRepo->findOneById($id);
```

Comment:
why are you doing so many database calls? why not retrieve all the payment methods first?